web-marketplace/src/pages/BuyCredits/BuyCredits.Form.tsx

```diff
@@ -205,8 +220,66 @@ export const BuyCreditsForm = ({
     ],
   );
 
+  const goToPaymentInfo = useCallback(
+    () => handleActiveStep(1),
+    [handleActiveStep],
+  );
+
+  const allowedDenoms = allowedDenomsData?.allowedDenoms;
```

Comment:
We could avoid re-renders in child components by memoizing `allowedDenoms` with `useMemo`.